stdlib/public/SDK/Foundation/TypePreservingNSNumber.mm

```diff
@@ -153,12 +257,17 @@ - (C_TYPE)METHOD_NAME { \
   }
 
 DEFINE_ACCESSOR(char, charValue)
+DEFINE_ACCESSOR(unsigned char, unsignedCharValue)
+DEFINE_ACCESSOR(short, shortValue)
+DEFINE_ACCESSOR(unsigned short, unsignedShortValue)
 DEFINE_ACCESSOR(int, intValue)
 DEFINE_ACCESSOR(unsigned int, unsignedIntValue)
 DEFINE_ACCESSOR(long long, longLongValue)
 DEFINE_ACCESSOR(unsigned long long, unsignedLongLongValue)
 DEFINE_ACCESSOR(float, floatValue)
 DEFINE_ACCESSOR(double, doubleValue)
+DEFINE_ACCESSOR(NSInteger, integerValue)
+DEFINE_ACCESSOR(NSUInteger, unsignedIntegerValue)
```

Comment:
Yep, and the default implementations in the `NSNumber` root class just infinitely looped trying to redispatch themselves. Fun.